File: Arcade/Intro/2 - Edge of the Ocean/6 - makeArrayConsecutive2/makeArrayConsecutive2.rb
def makeArrayConsecutive2(statues)
    min = 9999
    max = -1
    
    statues.each { |num|
        update_min = num < min
        update_max = num > max
        
        if(update_min)
            min = num
        end
        if(update_max)
            max = num
        end
    }
    
    return (max-min)-(statues.length-1)
end

File: Arcade/Intro/2 - Edge of the Ocean/6 - makeArrayConsecutive2/makeArrayConsecutive2.cpp
int makeArrayConsecutive2(std::vector<int> statues) {

    int min = 9999;
    int max = -1;
    
    for(std::vector<int>::iterator it = statues.begin(); it != statues.end(); it++)
    {
        bool updateMin = *it < min;
        bool updateMax = *it > max;
        
        if(updateMax) max = *it;
        if(updateMin) min = *it;
    }
    
    int result = (max - min) - (statues.size()-1);
    
    return result;
}
